test/framework/filetools.py

```diff
@@ -189,11 +188,14 @@ def test_download_file(self):
 
         # install broken proxy handler for opening local files
         # this should make urllib2.urlopen use this broken proxy for downloading from a file:// URL
-        proxy_handler = urllib2.ProxyHandler({'file': 'file://%s/nosuchfile' % test_dir})
+        proxy_handler = urllib2.ProxyHandler({'https': 'http://%s/nosuchfile' % test_dir})
         urllib2.install_opener(urllib2.build_opener(proxy_handler))
 
         # downloading over a broken proxy results in None return value (failed download)
         # this tests whether proxies are taken into account by download_file
+        fn = "robots.txt"
+        source_url = "https://jenkins1.ugent.be/"
+        target_location = os.path.join(self.test_buildpath, 'some', 'subdir', fn)
         self.assertEqual(ft.download_file(fn, source_url, target_location), None, "download over broken proxy fails")
```

Comment:
I was purposely using a broken proxy for `file://`, since then this test doesn't rely on whether or not you're online.

If you're offline, trying to download `robots.txt` from (wherever) will also fail and return `None`, but then you may be no longer checking the broken proxy aspect...

I know that installing a proxy for `file://` is utterly stupid, but it does the job for testing whether using a broken proxy actually makes things fail...